Self_Driving_Car_Engineer/Capstone_Project/CarND-Capstone/ros/src/tl_detector/light_classification/tl_classifier.py

```diff
@@ -110,49 +136,28 @@ def get_classification(self, image):
 
         Returns:
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
-
         """
-        #image_np = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
-        
-        
         image_np = np.expand_dims(np.asarray(image, dtype=np.uint8), 0)
-        
-        with tf.Session(graph=self.detection_graph) as self.sess:                
+
+        with tf.Session(graph=self.detection_graph) as self.sess:
             # Actual detection.
-            (boxes, scores, classes) = self.sess.run([self.detection_boxes, self.detection_scores, self.detection_classes], 
-                                        feed_dict={self.image_tensor: image_np})
+            (boxes, scores, classes) = self.sess.run([self.detection_boxes, self.detection_scores, self.detection_classes],
+                                                     feed_dict={self.image_tensor: image_np})
 
             boxes = np.squeeze(boxes)
             scores = np.squeeze(scores)
             classes = np.squeeze(classes)
-            confidence_cutoff = 0.1
-        
-            # Filter boxes with a confidence score less than `confidence_cutoff`
-            final_boxes, final_scores, final_classes = self.filter_boxes(confidence_cutoff, boxes, scores, classes)
-            #print(final_classes, final_scores)
-	    #print(final_classes.size)
-
-            # Enable for Visualization
-	    if self.enable_visualization and len(final_boxes) > 0 :
-		width, height, channels = image.shape
-            	box_coords = self.to_image_coords(final_boxes, height, width)
-            	image = self.draw_boxes(image, box_coords, final_classes)
-	    	#plt.figure(figsize=(12, 8))
-            	#plt.imshow(image)
-            
         
-        if final_classes.size != 0 and final_scores[0] is not None and final_scores[0] > confidence_cutoff:
-            class_name = self.category_index[final_classes[0]]['name']
-            #print((class_name, final_scores[0]))
-            #rospy.logwarn("final_class_prediction={}".format(final_classes[0]))
-            if final_classes[0] == 1:
-                return TrafficLight.GREEN
-            elif final_classes[0] == 2:
-                return TrafficLight.YELLOW
-            elif final_classes[0] == 3:
-                return TrafficLight.RED
-            
-        
-        return TrafficLight.UNKNOWN
+            # Filter boxes with a confidence score less than CONFIDENCE_CUTOFF
+            final_boxes, final_scores, final_classes = self.filter_boxes(CONFIDENCE_CUTOFF, boxes, scores, classes)
+
+            if len(final_boxes) > 0:
+                state, final_boxes, final_classes = self.find_state_change_and_noise_filter(final_boxes, final_classes)
```

Comment:
where do we use this state?

Reply:
good point. I was using that in the notebook. Now its not required

Reply:
@arunsuresh16